src/frontend/src/lib/utils/auth.utils.ts

```diff
@@ -1,20 +1,33 @@
+import { idbStorage } from '$lib/api/idb-auth.api';
 import { AUTH_ALTERNATIVE_ORIGINS, AUTH_DERIVATION_ORIGIN } from '$lib/constants/app.constants';
 import { isNullishOrEmpty } from '$lib/utils/input.utils';
 import type { Identity } from '@dfinity/agent';
-import { AuthClient } from '@dfinity/auth-client';
+import { AuthClient, KEY_STORAGE_DELEGATION } from '@dfinity/auth-client';
 import { notEmptyString } from '@dfinity/utils';
 
 export const createAuthClient = (): Promise<AuthClient> =>
 	AuthClient.create({
+		storage: idbStorage,
 		idleOptions: {
 			disableIdle: true,
 			disableDefaultIdleCallback: true
 		}
 	});
 
+/**
+ * When the user signs out, we create a new AuthClient.
+ * Since agent-js persists identity keys in IndexedDB by default,
+ * if the old keys remain in storage, they could be tampered with and affect the next login.
+ * To ensure each session starts clean and safe, we clear the stored keys before creating a new AuthClient.
+ */
+export const safeCreateAuthClient = async (): Promise<AuthClient> => {
+	await idbStorage.remove(KEY_STORAGE_DELEGATION);
```

Comment:
What's important is to clean the session key rather than the delegation, because that's the one that signs messages.
```suggestion
	await idbStorage.remove(KEY_STORAGE_KEY);
```